### PR TITLE
Fix leaky tests.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/flutter/test/painting/image_stream_test.dart [hunk 1]
```diff
@@ -293,7 +296,9 @@ void main() {
       scale: 1.0,
     );
 
-    void listener(ImageInfo image, bool synchronousCall) { }
+    void listener(ImageInfo image, bool synchronousCall) {
+      addTearDown(image.dispose);
+    }
     imageStream.addListener(ImageStreamListener(listener));
```
Comment (Renzo-Olivares) [inline packages/flutter/test/painting/image_stream_test.dart:302]: should we also remove this listener at the end of this test?

File: packages/flutter/test/painting/image_stream_test.dart [hunk 2]
```diff
@@ -107,7 +104,9 @@ void main() {
     await tester.idle();
     expect(mockCodec.numFramesAsked, 0);
 
-    void listener(ImageInfo image, bool synchronousCall) { }
+    void listener(ImageInfo image, bool synchronousCall) {
+      addTearDown(image.dispose);
+    }
     imageStream.addListener(ImageStreamListener(listener));
```
Comment (Renzo-Olivares) [inline packages/flutter/test/painting/image_stream_test.dart:110]: should we also remove this listener at the end of the test?

File: packages/flutter/test/painting/image_stream_test.dart (hunk 1 above)
Comment (polina-c) [reply packages/flutter/test/painting/image_stream_test.dart:302]: For some reasons this one does not cause leaks if not removed.

File: packages/flutter/test/painting/image_stream_test.dart (hunk 2 above)
Comment (polina-c) [reply packages/flutter/test/painting/image_stream_test.dart:110]: For some reasons this one does not cause leaks if not removed.

**Review by Renzo-Olivares — APPROVED** [review]
LGTM, thanks for the fix

**Pull request merged**